logcrash: `TestCrashReportingSafeError` fails on `s390x`

**[How to gain access to an s390x machine for testing (Internal link)](https://cockroachlabs.atlassian.net/wiki/spaces/devinf/pages/4536467683/Reproducing+an+issue+on+s390x)**

```
* ERROR: a panic has occurred!
* panic: boom
* (1) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/log/logcrash_test.TestCrashReportingPacket
*   |   crash_reporting_packet_test.go:1059
*   | testing.tRunner
*   |   GOROOT/src/testing/testing.go:1690
*   | runtime.goexit
*   |   src/runtime/asm_s390x.s:797
* Wraps: (2) panic: boom
* Error types: (1) *withstack.withStack (2) *errutil.leafError
*
*
* ERROR: Queued as error 2ecaf3d13fc2485c88e02a38a6ef22d6
*
*
* ERROR: a panic has occurred!
* panic: baam
* (1) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/util/log/logcrash_test.TestCrashReportingPacket
*   |   crash_reporting_packet_test.go:1067
*   | testing.tRunner
*   |   GOROOT/src/testing/testing.go:1690
*   | runtime.goexit
*   |   src/runtime/asm_s390x.s:797
* Wraps: (2) panic: baam
* Error types: (1) *withstack.withStack (2) *errutil.leafError
*
*
* ERROR: Queued as error fd07aa8679974c75b4bbb641576fb082
*
*
* ERROR: Queued as error 56da26d2ebf44f8da1197912de4d0658
*
--- FAIL: TestCrashReportingSafeError (0.00s)
    --- FAIL: TestCrashReportingSafeError/safeErr#02 (0.00s)
        crash_reporting_test.go:1298: Diff:
            --- Expected
            +++ Actual
            @@ -14,3 +14,3 @@
               | runtime.goexit
            -  |        ...asm_amd64.s:NN
            +  |        ...asm_s390x.s:NN
             Wraps: (2) some visible detail
    --- FAIL: TestCrashReportingSafeError/safeErr#03 (0.00s)
        crash_reporting_test.go:1298: Diff:
            --- Expected
            +++ Actual
            @@ -14,3 +14,3 @@
               | runtime.goexit
            -  |        ...asm_amd64.s:NN
            +  |        ...asm_s390x.s:NN
             Wraps: (2) secondary error attachment
    --- FAIL: TestCrashReportingSafeError/safeErr#05 (0.00s)
        crash_reporting_test.go:1298: Diff:
            --- Expected
            +++ Actual
            @@ -14,3 +14,3 @@
               | runtime.goexit
            -  |        ...asm_amd64.s:NN
            +  |        ...asm_s390x.s:NN
             Wraps: (2) I like A and my pin code is × or 9999
    --- FAIL: TestCrashReportingSafeError/safeErr#06 (0.00s)
        crash_reporting_test.go:1298: Diff:
            --- Expected
            +++ Actual
            @@ -14,3 +14,3 @@
               | runtime.goexit
            -  |        ...asm_amd64.s:NN
            +  |        ...asm_s390x.s:NN
             Wraps: (2) this is preserved: 6
    --- FAIL: TestCrashReportingSafeError/safeErr#09 (0.00s)
        crash_reporting_test.go:1298: Diff:
            --- Expected
            +++ Actual
            @@ -46,3 +46,3 @@
               | runtime.goexit
            -  |        ...asm_amd64.s:NN
            +  |        ...asm_s390x.s:NN
             Wraps: (8) ×
    --- FAIL: TestCrashReportingSafeError/safeErr#11 (0.00s)
        crash_reporting_test.go:1298: Diff:
            --- Expected
            +++ Actual
            @@ -14,3 +14,3 @@
               | runtime.goexit
            -  |        ...asm_amd64.s:NN
            +  |        ...asm_s390x.s:NN
             Wraps: (2) secondary error attachment
            @@ -49,3 +49,3 @@
               |   | runtime.goexit
            -  |   |    ...asm_amd64.s:NN
            +  |   |    ...asm_s390x.s:NN
               | Wraps: (6) ×
```

Epic: CRDB-21133

Jira issue: CRDB-50329

--- a/pkg/util/log/logcrash/crash_reporting_test.go
+++ b/pkg/util/log/logcrash/crash_reporting_test.go
@@ -317,6 +317,7 @@ func TestCrashReportingSafeError(t *testing.T) {
 			errStr = fileref.ReplaceAllString(errStr, "...$2:NN")
 			errStr = strings.ReplaceAll(errStr, "asm_arm64.s", "asm_amd64.s")
 			errStr = strings.ReplaceAll(errStr, "asm_ppc64x.s", "asm_amd64.s")
+			errStr = strings.ReplaceAll(errStr, "asm_s390x.s", "asm_amd64.s")
 			if errStr != test.expErr {
 				diff, _ := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
 					A:        difflib.SplitLines(test.expErr),

